app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -111,6 +113,60 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = 0
     )
 
+    private var experimentDefaultBrowserLastTappedExperiment by stringPreference(
+        appContext.getPreferenceKey(R.string.pref_key_experiment_default_last_tapped_experiment),
+        default = ""
+    )
+
+    private var experimentDefaultBrowserLastTappedTimeInMillis by longPreference(
+        appContext.getPreferenceKey(R.string.pref_key_experiment_default_last_tapped_time_in_millis),
+        default = -1
+    )
+
+    private var experimentDefaultBrowserWasDefaultBeforeTry by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_experiment_default_browser_was_default),
+        default = false
+    )
+
+    /*
+    * As there is not a consistent way to accross OS version to determine that the default browser
+    * was updated as result of tapping of one of the experiment, we are using some heuristics,
+    * to determine it.
+    * 1) The user has previously tapped one of the experiments.
+    * 2) The actual default browser it's Firefox.
+    * 3) Firefox was not the default browser before the tap.
+    * 4) The default browser was updated interval of 5 minutes of the last tap on a experiment.
+    * */
+    @Suppress("MagicNumber")
+    fun wasFirefoxSetAsDefaultByExperiment(experiment: Event.ExperimentDefaultBrowser.Type): Boolean {
+        val minuteInMS = 60 * 1000
+        val maxAllowedTime = 5 * 1000 // Max time between the last intent of setting a default browser
+        val browsers = BrowsersCache.all(appContext)
+        val wasSameExperiment = experimentDefaultBrowserLastTappedExperiment == experiment.name
+        val wasRecentlyUpdated =
+            experimentDefaultBrowserLastTappedTimeInMillis < Date().time - maxAllowedTime * minuteInMS
+        val wasDefaultBrowserBefore = experimentDefaultBrowserWasDefaultBeforeTry
+
+        // We want to determine if the previous intent of setting a default browser was successful
+        // and was produced by this experiment.
+
+        return browsers.isDefaultBrowser && !wasDefaultBrowserBefore && wasSameExperiment && wasRecentlyUpdated
+    }
+
+    fun recordIntentOfSettingDefaultBrowserExperiment(experiment: Event.ExperimentDefaultBrowser.Type) {
+        val browsers = BrowsersCache.all(appContext)
+
+        experimentDefaultBrowserLastTappedExperiment = experiment.name
+        experimentDefaultBrowserLastTappedTimeInMillis = Date().time
+        experimentDefaultBrowserWasDefaultBeforeTry = browsers.isDefaultBrowser
+    }
```

Comment:
@mcarare I think this could be useful for other experiment, for indicating if the intent as setting a default browser was successful and indicate which experiment branch was the source of it.

Reply:
I am not sure we need `wasDefaultBrowserBefore`. Because we do not show experiment cards or options when it is already set as default.

Reply:
Yeah, that's right I will remove this one 👍🏽